plugins/woocommerce/src/Internal/ComingSoon/ComingSoonRequestHandler.php

```diff
@@ -48,13 +51,21 @@ public function handle_template_include( $template ) {
 		// A coming soon page needs to be displayed. Don't cache this response.
 		nocache_headers();
 
+		$is_fse_theme         = wc_current_theme_is_fse_theme();
+		$is_store_coming_soon = $this->coming_soon_helper->is_store_coming_soon();
+
+		if ( ! $is_fse_theme ) {
```

Comment:
This should probably be the opposite of the check [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Blocks/Domain/Bootstrap.php#L134), no? Otherwise, in a classic theme with support for block templates we are initializing those classes twice.

Reply:
You're right! I've updated the logic in [d655c60](https://github.com/woocommerce/woocommerce/pull/50507/commits/d655c60747bb7f9d9d246e17f80f70de24bc4733).

I've tested registration with block theme, classic theme, and classic theme with blocks support using both [storefront-child-i.zip](https://github.com/user-attachments/files/16054910/storefront-child-i.zip) and [storefront-child-ii.zip](https://github.com/user-attachments/files/16054901/storefront-child-ii.zip) and it seems like the init runs only once in coming soon mode after the change.

Reply:
Documenting the behaviour:

- twentytwentyfour: Initialize at bootstrap
- storefront-child-i.zip: Initialize at bootstrap
- storefront-child-ii.zip: Initialize at bootstrap
- storefront: Initialize at coming soon